src/components/calcite-flow/calcite-flow.tsx

```diff
@@ -26,6 +26,34 @@ export class CalciteFlow {
    */
   @Prop({ reflect: true }) theme: CalciteTheme;
 
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Removes the currently active `calcite-flow-item`.
+   */
+  @Method()
+  async back(): Promise<HTMLCalciteFlowItemElement> {
+    const lastItem = this.el.querySelector(
+      "calcite-flow-item:last-child"
+    ) as HTMLCalciteFlowItemElement;
+
+    if (!lastItem) {
+      return;
+    }
+
+    const beforeBack = lastItem.beforeBack ? lastItem.beforeBack : () => Promise.resolve();
+
+    await beforeBack.call(lastItem);
+
+    lastItem.remove();
+
+    return lastItem;
```

Comment:
We may want to return the promise here. That allows anyone calling back to be able to fall into a `.catch()` for error handling should the custom promise they passed error.
```
return lastItem.beforeBack().then(() => {
  lastItem.remove();
});
```

Reply:
This is an async function so a promise is always returned.

Reply:
Yes, but we aren't returning the beforeBack promise. If that Promise were to error, it would get swallowed by this code. I'm suggesting preserving the Promise chain.

```
const flow = document.querySelector("calcite-flow");
flow.back().catch(error => {
  // beforeBack must have errored, handle gracefully
}).then(() => {
  // resume operations
});

See the fiddle how in one case the catch works and in the other an exception is thrown.
https://jsfiddle.net/pr3tori4n/834auzyd/

Reply:
Fixed. thanks!